src/components/LatestDataTable.tsx

```diff
@@ -0,0 +1,203 @@
+import Button from "./commons/Button";
+import GradientCardContainer from "./commons/GradientCardContainer";
+import LinkText from "./commons/LinkText";
+
+type DataType = "blocks" | "transactions";
+
+export interface TxnWalletInfo {
+  from: string;
+  to: string;
+}
+
+export interface BlockInfo {
+  transactionsPerBlock: string;
+  blockTimeInSec: string;
+}
+
+export interface RowData {
+  id: string;
+  transactionId: string;
+  tokenAmount: string;
+  txnOrBlockInfo: TxnWalletInfo | BlockInfo;
+  datetime: string;
+}
+
+interface Props {
+  type: DataType;
+  title: string;
+  data: RowData[];
+  containerClass?: string;
+  amountPrefix?: string;
+}
+
+export default function LatestDataTable({
+  type,
+  title,
+  data,
+  amountPrefix = "",
+  containerClass = "",
+}: Props): JSX.Element {
+  return (
+    <div
+      data-testid={`latest-${type}-table`}
+      className={`${containerClass ?? ""}`}
+    >
+      <GradientCardContainer>
+        <div className="p-5 py-6 md:p-10">
+          <div className="flex flex-col md:grid md:grid-cols-6 md:items-center">
+            <h2
+              data-testid={`latest-${type}-title`}
+              className="text-white-50 font-bold text-xl md:text-2xl leading-8 pb-4 md:pb-0 md:col-span-3"
+            >
+              {title}
+            </h2>
+            <div className="md:order-last md:flex-1 md:pt-6 md:col-span-8">
+              {data.map((row, index) => (
+                <RowItem
+                  key={row.id}
+                  rowIndex={index}
+                  type={type}
+                  rowData={row}
+                  amountPrefix={amountPrefix}
+                />
+              ))}
+            </div>
+            <div className="pt-5 md:pt-0 md:col-span-3 md:col-end-auto">
+              <div className="md:flex md:justify-end md:text-rightt">
+                <Button
+                  testId={`view-${type}`}
+                  label={`View all ${type}`}
+                  href={`/${type}`}
+                  customStyle="w-full md:w-auto"
+                />
+              </div>
+            </div>
+          </div>
+        </div>
+      </GradientCardContainer>
+    </div>
+  );
+}
+
+const truncateTextFromMiddle = (text: string, length = 5): string => {
+  const truncatedText = `${text.substring(0, length)}...${text.substring(
+    text.length - length,
+    text.length
+  )}`;
+  return truncatedText;
+};
+
+const DmxTokenSymbol = "DMXTc";
```

Comment:
let's refactor this, being used in `TokenStatsDisplay` too

Reply:
Updated, moved to `constants.ts` file